app/feature/insurances/src/test/kotlin/com/hedvig/android/feature/insurances/insurance/presentation/InsurancePresenterTest.kt

```diff
@@ -0,0 +1,263 @@
+package com.hedvig.android.feature.insurances.insurance.presentation
+
+import app.cash.turbine.Turbine
+import arrow.core.Either
+import arrow.fx.coroutines.raceN
+import assertk.assertAll
+import assertk.assertThat
+import assertk.assertions.containsSubList
+import assertk.assertions.isEqualTo
+import assertk.assertions.isFalse
+import assertk.assertions.isTrue
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.feature.insurances.data.GetCrossSellsUseCase
+import com.hedvig.android.feature.insurances.data.GetInsuranceContractsUseCase
+import com.hedvig.android.feature.insurances.data.InsuranceContract
+import com.hedvig.android.molecule.test.test
+import com.hedvig.android.notification.badge.data.crosssell.card.FakeCrossSellCardNotificationBadgeService
+import giraffe.type.TypeOfContract
+import kotlinx.coroutines.test.runTest
+import octopus.CrossSalesQuery
+import org.junit.Assert
+import org.junit.Test
+
+internal class InsurancePresenterTest {
```

Comment:
Please do give this class a nice look, I feel like here is where we get the biggest benefit of using Molecule.
The tests read so nicely like:
Start the test -> see initial state
Do something -> See that the UI is in the appropriate state
Make the backend return the response -> Automatically the UI has already reacted to it.

It just feels so easy to reason about, especially without having to know about the inner workings of how the ViewModel is doing things and when you need to `advanceUntilIdle()` or not.

Reply:
Looks super good